sdk/core/core-util/src/aborterUtils.ts

```diff
@@ -61,3 +71,23 @@ export function createAbortablePromise<T>(
     abortSignal?.addEventListener("abort", onAbort);
   });
 }
+
+/**
+ * promise.race() implementation that aborts losers as soon as the first promise fulfills.
+ */
+export async function racePromisesAndAbortLosers<T>(
+  promises: ((abortOptions: AbortOptions) => Promise<T>)[],
+  abortSignal?: AbortSignalLike
+): Promise<T> {
+  const loserAborter = new AbortController();
+  const loserAbortListener = () => {
+    loserAborter.abort();
+  };
+  abortSignal?.addEventListener("abort", loserAbortListener);
```

Comment:
nitpick: should this be registered after the promise has been created?

Reply:
the call right after this is `Promise.race()`, so, it is good?